front/src/modules/ui/table/components/EntityTableColumnMenu.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { ComponentProps, useRef } from 'react';
+import { useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+
+import { IconButton } from '@/ui/button/components/IconButton';
+import { IconButtonGroup } from '@/ui/button/components/IconButtonGroup';
+import { DropdownMenu } from '@/ui/dropdown/components/DropdownMenu';
+import { DropdownMenuItem } from '@/ui/dropdown/components/DropdownMenuItem';
+import { DropdownMenuItemsContainer } from '@/ui/dropdown/components/DropdownMenuItemsContainer';
+import { IconPlus } from '@/ui/icon';
+import { useListenClickOutside } from '@/ui/utilities/pointer-event/hooks/useListenClickOutside';
+
+import type {
+  ViewFieldDefinition,
+  ViewFieldMetadata,
+} from '../types/ViewField';
+
+const StyledColumnMenu = styled(DropdownMenu)`
+  font-weight: ${({ theme }) => theme.font.weight.regular};
+`;
+
+const StyledIconButtonGroup = styled(IconButtonGroup)`
+  display: none;
+  position: absolute;
+  right: ${({ theme }) => theme.spacing(1)};
+`;
+const styledIconButtonGroupClassName = 'column-menu-item-icon-button-group';
+
+const StyledColumnMenuItem = styled(DropdownMenuItem)`
+  position: relative;
+
+  &:hover {
+    .${styledIconButtonGroupClassName} {
+      display: flex;
+    }
+  }
+`;
+
+type EntityTableColumnMenuProps = {
```

Comment:
I usually prefix the component name to the type name in case we need to export the Props type to reuse it later (in stories, in a parent component...). But if the Props are kept locally then I don't really mind using `OwnProps`.

Reply:
I actually like your convention more, it looks more flexible. The only drawback is that when we rename a component, we have one more renaming to do but with IDEs capabilities this should not be a problem. Let's leave it like this and we will progressively replace the OwnProps. I will also add it to the doc